proxy-node-gateway/src/main/java/uk/gov/ida/notification/resources/HubResponseResource.java

```diff
@@ -47,32 +45,42 @@ public View hubResponse(
         @FormParam("RelayState") String relayState,
         @Session HttpSession session) {
 
-        String hubRequestId = session.getAttribute(SESSION_KEY_HUB_REQUEST_ID).toString();
-        String eidasRequestId = session.getAttribute(SESSION_KEY_EIDAS_REQUEST_ID).toString();
-        String connectorEncrpytionCredential = session.getAttribute(SESSION_KEY_EIDAS_CONNECTOR_PUBLIC_CERT).toString();
-        String connectorNodeUrl = session.getAttribute(SESSION_KEY_EIDAS_DESTINATION).toString();
-        String eidasRelayState = session.getAttribute(SESSION_KEY_EIDAS_RELAY_STATE).toString();
+        GatewaySessionData sessionData = GatewaySessionDataValidator.getValidatedSessionData(session);
 
-        LOG.info(String.format("[HUB Response] received for hub authn request ID '%s', eIDAS authn request ID '%s'", hubRequestId, eidasRequestId));
+        LOG.info(
+            String.format(
+                "[HUB Response] received for session '%s', hub authn request ID '%s', eIDAS authn request ID '%s'",
+                session.getId(),
+                sessionData.getHubRequestId(),
+                sessionData.getEidasRequestId()
+            )
+        );
 
         HubResponseTranslatorRequest translatorRequest = new HubResponseTranslatorRequest(
             hubResponse,
```

Comment:
Lets pass `sessionData` to `HubResponseTranslatorRequest` as a object rather and supplying the strings.

Reply:
yep, will do.

Reply:
@tomrosier I'm just looking at passing the session data object to the hub response translator request's constructor. It'll mean moving the session data class to shared, as that's where the hub response translator request lives. Feels like a weird thing to do. What do you think?

Reply:
Ah, yeah I can see now why its that way, I thought it was specific to the gateway. Leave it as is.